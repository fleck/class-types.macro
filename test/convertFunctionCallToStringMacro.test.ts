import pluginTester from "babel-plugin-tester";
import plugin from "babel-plugin-macros";

pluginTester({
  plugin,
  babelOptions: { filename: __filename, presets: ["@babel/preset-react"] },
  tests: {
    "fail if not a function": {
      error: true,
      code: `
      import ct from '../../ct.macro';

      ct({someOption: true}, \`
        some stuff
      \`)
    `,
    },
    "transform function call to string": {
      code: `
        import ct from '../../ct.macro';
  
        ct("class", "other-class")
      `,
      output: '"class other-class";',
    },
    "transform function call with single argument identifier to just an identifier": {
      code: `
        import ct from '../../ct.macro';

        const style = ct("class", "other-class");
  
        ct(style);
      `,
      output: `const style = "class other-class";
style;`,
    },
    "transform a function call with 2 identifiers to a binary expression": {
      code: `
        import ct from '../../ct.macro';

        const style = ct("class", "other-class");

        const style2 = ct("other");
  
        ct(style, style2);
      `,
      output: `const style = "class other-class";
const style2 = "other";
style + (" " + style2);`,
    },
    "transform jsx string": {
      code: `
          import ct from '../../ct.macro';
    
          <div className={ct("class", "other-class")} />;
        `,
      snapshot: true,
    },
  },
});
