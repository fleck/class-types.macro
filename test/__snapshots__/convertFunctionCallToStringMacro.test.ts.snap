// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros transform jsx string: transform jsx string 1`] = `

import ct from '../../class-types.macro';
    
<div className={ct("class", "other-class")} />;

      ↓ ↓ ↓ ↓ ↓ ↓

/*#__PURE__*/
React.createElement("div", {
  className: "class other-class",
});


`;

exports[`macros transform jsx with variables and function call to ct: transform jsx with variables and function call to ct 1`] = `

import ct from '../../class-types.macro';

const style = ct("class", "other-class");
    
<div className={ct("string", style, ct("hey", style))} />;

      ↓ ↓ ↓ ↓ ↓ ↓

const style = "class other-class";

/*#__PURE__*/
React.createElement("div", {
  className: "string" + (" " + (style + (" " + ct("hey", style)))),
});


`;

exports[`macros transform jsx with variables and function call: transform jsx with variables and function call 1`] = `

import ct from '../../class-types.macro';

const style = ct("class", "other-class");
    
<div className={ct("string", style, getStyles())} />;

      ↓ ↓ ↓ ↓ ↓ ↓

const style = "class other-class";

/*#__PURE__*/
React.createElement("div", {
  className: "string" + (" " + (style + (" " + getStyles()))),
});


`;

exports[`macros transform jsx with variables and ternary: transform jsx with variables and ternary 1`] = `

import ct from '../../class-types.macro';

const style = ct("class", "other-class");
    
<div className={ct("string", style, true ? "this" : "that")} />;

      ↓ ↓ ↓ ↓ ↓ ↓

const style = "class other-class";

/*#__PURE__*/
React.createElement("div", {
  className: "string" + (" " + (style + (" " + (true ? "this" : "that")))),
});


`;

exports[`macros transform jsx with variables: transform jsx with variables 1`] = `

import ct from '../../class-types.macro';

const style = ct("class", "other-class");
    
<div className={ct("string", style)} />;

      ↓ ↓ ↓ ↓ ↓ ↓

const style = "class other-class";

/*#__PURE__*/
React.createElement("div", {
  className: "string" + (" " + style),
});


`;
